source/bayes_simulation.Rmd

```diff
@@ -45,112 +44,268 @@ correctly identifies 80 percent as "OK" but says that 20 percent are
+N = 1000000  # number of cars
 
--OR-
+# Generate an array with as many entries as there are cars, each
+# being either 'working' or 'faulty'
+cars = np.random.choice(['working', 'faulty'], p=[0.7, 0.3], size=N)
 
-1.  Choose a number randomly between "1" and "100."
+# Count how many cars are working and how many are faulty
+N_working = np.count_nonzero(cars == 'working')
+N_faulty = N - N_working
 
-2.  If "28-30" or "31-45:, record; otherwise draw another number, and
-    repeat until a number is recorded.
+# Create a new array in which to stores what a mechanic says
+# about the car: 'approved' or 'not approved'
+mechanic_says = np.empty_like(cars, dtype=object)
 
-3.  Repeat step 2 and count the proportion "28-30" among the total
-    "28-45."
+# For each working car, generate a mechanic mechanic_says/declared faulty
+# label with the given probabilities
+mechanic_says[cars == 'working'] = np.random.choice(
+    ['approved', 'not approved'], p=[0.8, 0.2], size=N_working
+)
 
-The key modeling step is excluding a trial from consideration (without
-making any record of it) if it produces an irrelevant observation, and
-continuing to do so until the process produces an observation of the
-sort about which you are presently inquiring.
+# For each faulty car, generate a mechanic mechanic_says/declared faulty
+# label with the given probabilities
+mechanic_says[cars == 'faulty'] = np.random.choice(
+    ['approved', 'not approved'], p=[0.1, 0.9], size=N_faulty
+)
 
-Using RESAMPLING STATS, an answer may be produced as follows:
+# Identify all cars that were approved
+approved = (mechanic_says == 'approved')
 
-"01 - 27" = actually faulty, said to be "faulty" "28 - 30 = faulty, said
-to be "OK"
+# Identify cars that are faulty AND were approved
+faulty_but_approved = (cars == 'faulty') & approved
 
-"31 - 86" = OK, "OK"
+# Count the number of cars that are faulty but approved, as well as
+# the total number of cars that were approved
+N_faulty_but_approved = np.count_nonzero(faulty_but_approved)
+N_approved = np.count_nonzero(approved)
 
-"87 - 100" = OK, "faulty"
+# Calculate the ratio, which is the answer we seek
+k = N_faulty_but_approved / N_approved
 
-`           REPEAT 1000         `
+print(f'{k * 100:.2}%')
+```
 
-Do 1000 repetitions
+There is about a 5% chance of a car being broken after it has been
+approved by a mechanic.
 
-`           GENERATE 1 1,100 a         `
+### Calculation without simulation
 
-Generate a number between "1" and "100"
+Often, we use simulations to give us a sense of what the right answer
+to a problem should be. Thereafter, we can look into different—sometimes more elegant or accurate—ways of modeling and solving it.
 
-`           IF a between 28 86         `
+Let's examine the following tree diagram of our car selection:
 
-If it's between "28" and "86" (those that say "good")
+![](diagrams/car-tree.png)
 
-`           SCORE a z         `
+We see that there are two paths, highlighted, that results in a car
```

Comment:
"paths" and "tree" don't match?

Maybe something on the lines of:

In the long run, in our simulation, about 70% of the cars will end with the label "working", and about 30% will end up with the label "faulty".   We just took 10000 sample cars above, bu in fact, the larger the number of samples cars we take, the closer we will get to 70% "working" and 30% "faulty". So, with many samples, we can think of 70% of these samples flowing down the "working" path, and 30% flowing along the "faulty" path.

You've done this below - I see - but maybe better here ...